exporters/failure/collector_jira.py

```diff
@@ -14,6 +15,17 @@ class JiraFailureCollector(AbstractFailureCollector):
     """
 
     def __init__(self, user, apikey, server):
+        if not os.environ.get("PROJECT"):
+            logging.info(
+                "Missing Project Field Parameter, querying all projects."
+            )
+            self.project_field = None
+        else:
+            logging.info(
+                "Querying issues from only '%s' projects.",
+                os.environ.get("PROJECT"),
+            )
+            self.project_field = os.environ.get("PROJECT")
```

Comment:
We typically try to read all configuration within the exporter app.py itself, and pass that in to the class via a parameter-- it makes testing easier. But this is more or less the pattern we'd want to follow.

Reply:
As I could not find a pattern about that, so I review the Service Now Collector and replicated the same structure from the [collector_servicenow.py](https://github.com/konveyor/pelorus/blob/master/exporters/failure/collector_servicenow.py#L24) file.

In both cases make sense to initialize the class with the specific parameters used, otherwise the constructor could be so complex to manage the different properties (requiered and optional).

WDYT?

Reply:
@rmarting the pattern we would recommend would be to follow something similar to what we did for namespaces in the committime exporter: https://github.com/konveyor/pelorus/blob/master/exporters/committime/app.py#L65-L73

Also, since PROJECT wasn't used anyway, let's go ahead and make it the plural, PROJECTS so that we've got more correct nomenclature.

Reply:
@etsauer @KevinMGranger Refactored following your recommendations. Hope now it is aligned with the patterns wanted.

Reply:
Holding off on approving until we can test more but it looks good overall.